pkg/services/ngalert/state/manager.go

```diff
@@ -413,7 +415,7 @@ func (st *Manager) staleResultsHandler(ctx context.Context, evaluatedAt time.Tim
 				s.EndsAt = evaluatedAt
 				s.Resolved = true
 				if st.historian != nil {
-					st.historian.RecordState(ctx, alertRule, s.Labels, evaluatedAt,
+					st.historian.RecordState(ctx, alertRule, s, evaluatedAt,
```

Comment:
Here if we drop `currentData` we can just set `s.State = eval.Normal` and `s.Reason = s.StateReason` to keep things consistent.